File: example/00001.py
# -*- coding: UTF-8 -*-
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash_table = {}
        for index,val in enumerate(nums):
            diff = target - val
            if diff in hash_table:
                return [hash_table[diff],index]
            hash_table[val] = index
        return []

File: example/00001.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
   hash_table := map[int]int{}
   for i, v := range nums {
       var diff int = target - v
       if p, ok := hash_table[diff]; ok {
           return []int{p, i}
       }
       hash_table[v] = i
   }
   return []int{}
}

func main() {
    var nums = []int{2,7,11,15}
    var target int  = 9
    result := twoSum(nums, target)
    fmt.Println(result)
}